pkg/operator/reports.go

```diff
@@ -91,8 +91,26 @@ func (op *Reporting) syncReport(logger log.FieldLogger, key string) error {
 		}
 		return err
 	}
+
 	sr := report.DeepCopy()
 
+	// check if the report is expired
+	if report.DeletionTimestamp == nil {
+		if reportExpired := isReportExpired(logger, report, time.Now()); reportExpired {
```

Comment:
I think I would like to see a single call site for handling an expired Report. WDYT about doing something like the following:
```go
err := op.handleExpiredReport(logger, sr, time.Now())
if err != nil {
    return err
}
```
and that `op.handleExpiredReport` would essentially encapsulate the work you're doing here:
```
func (op *Reporting) handleExpiredReport(logger logrus.FieldLogger, report *metering.Report, processTime time.Time) error {
    if report.DeletionTimestamp != nil {
       return fmt.Errorf("report was already marked for deletion")
    }
    // handle case where the report has expired, log message and exit early
    if reportHasExpired := isReportExpired(logger, report, processTime); !reportExpired {
        logger.Infof("the %s Report in the %s namespace has not reached the expiration data", report.Name, report.Namespace)
        return nil
    }
    // we know the report is considered "expired" by now
    if reportCanBeDeleted := op.isReportNotUsedAsInput(report); reportCanBeDeleted {
        // do report deletion work here
        ...
```

Reply:
Also if `isReportNotUsedAsInput` passes a reporting structure, I would expect that function to migrate to a method on the reporting type or be restructured such that it uses dependency injection to get the fields in the reporting structure you care about.

Reply:
After looking at some of the reporting-operator codebase, it looks like we have support for resolving the dependencies for a Report: https://github.com/kube-reporting/metering-operator/blob/1f5c0686730008b8bafcbc602514eb3120c132cd/pkg/operator/reports.go#L295-L301

That method on that interface would then return the following type:
```go
type DependencyResolutionResult struct {
	Dependencies *ReportQueryDependencies
	InputValues  map[string]interface{}
}
```

Reply:
encapsulated into function on Reporting type, testing functionality now.

Reply:
works again